test/providers/java_maven.test.js

```diff
@@ -25,7 +25,8 @@ suite('testing the java-maven data provider', () => {
 		"pom_deps_with_ignore_on_wrong",
 		"pom_deps_with_no_ignore",
 		"poms_deps_with_ignore_long",
-		"poms_deps_with_no_ignore_long"
+		"poms_deps_with_no_ignore_long",
+		"poms_deps_with_no_ignore_common_paths"
```

Comment:
@ruromero Need to rename it to "pom_deps_with_no_ignore_common_paths" ( actual directory name) , because test is failing as a result of this typo

Reply:
I added this recursion stop/halting condition to the code, because there are a lot of cases that there is a dependency with no continuing dependencies(even no direct ones) and it continues with no lines at all , or one empty lines ( the added halting condition)

```javascript
// exists condition
	if(lines.length === 0) {
		return
	}

// new condition in addition.
	if((lines.length === 1 && lines[0].trim() === ""))
	{
		return
	}
```

This one seems to solve majority of the failures, but we still have 2, one with mismatch of the content of sbom ( expected and actual) , and the other one that it inserted into infinity loop

Reply:
@ruromero So after invetigation that i've made, it turns out that setting on the verbose flag on big pom.xml files (with or without parents) causing the mvn:dependency plugin + goal to enter infinite loop, that's why the test already run for 30 minutes without progressing..

I Tested it myself , and it's indeed stuck on it, so eventually i gave up after couple of minutes
```shell
[zgrinber@zgrinber poms_deps_with_no_ignore_long]$ date
**Sun Sep  3 14:43:40 IDT 2023**
[zgrinber@zgrinber poms_deps_with_no_ignore_long]$ mvn  dependency:tree -DoutputType=text  -DoutputFile=file.output -Dverbose
[INFO] Scanning for projects...
[INFO] 
[INFO] --------------< com.redhat.examples:custom-metrics-demo >---------------
[INFO] Building custom-metrics-demo 0.0.1-SNAPSHOT
[INFO] --------------------------------[ jar ]---------------------------------
[INFO] 
[INFO] --- maven-dependency-plugin:3.2.0:tree (default-cli) @ custom-metrics-demo ---
[WARNING] The artifact xml-apis:xml-apis:jar:2.0.2 has been relocated to xml-apis:xml-apis:jar:1.0.b2
[WARNING] The artifact ehcache:ehcache:jar:1.2 has been relocated to net.sf.ehcache:ehcache:jar:1.2
[WARNING] The artifact xerces:xerces:jar:2.4.0 has been relocated to xerces:xercesImpl:jar:2.4.0
[WARNING] The artifact bouncycastle:bctsp-jdk14:jar:138 has been relocated to org.bouncycastle:bctsp-jdk14:jar:1.38
[WARNING] The artifact com.lowagie:itext:jar:4.2.0 has been relocated to com.itextpdf:itextpdf:jar:5.5.6: After release 2.1.7, iText moved from the MPLicense to the AGPLicense.
			The groupId changed from com.lowagie to com.itextpdf and the artifactId from itext to itextpdf.
			See http://itextpdf.com/functionalitycomparison for more information.
[WARNING] The artifact activation:activation:jar:1.0.2 has been relocated to javax.activation:activation:jar:1.0.2
[WARNING] The artifact jboss:javassist:jar:3.3.ga has been relocated to javassist:javassist:jar:3.3.GA
[WARNING] The artifact poi:poi:jar:2.5.1 has been relocated to poi:poi:jar:2.5.1-final-20040804
[WARNING] The artifact servletapi:servletapi:jar:2.3 has been relocated to javax.servlet:servlet-api:jar:2.3
[WARNING] The artifact javacc:javacc:jar:3.2 has been relocated to net.java.dev.javacc:javacc:jar:3.2
[WARNING] The artifact xstream:xstream:jar:1.2 has been relocated to com.thoughtworks.xstream:xstream:jar:1.2
[WARNING] The artifact xerces:xerces:jar:2.0.2 has been relocated to xerces:xercesImpl:jar:2.0.2
[WARNING] The artifact itext:itext:jar:1.3.1 has been relocated to com.lowagie:itext:jar:1.3.1
[WARNING] The artifact itext:itext:jar:1.3 has been relocated to com.lowagie:itext:jar:1.3
[WARNING] The artifact jstl:jstl:jar:1.0.6 has been relocated to javax.servlet:jstl:jar:1.0.6
[WARNING] The artifact xerces:xerces:jar:2.2.1 has been relocated to xerces:xercesImpl:jar:2.2.1
[WARNING] The artifact sun-jaf:activation:jar:1.0.2 has been relocated to javax.activation:activation:jar:1.0.2: This file has been relocated based on maven recommendations: http://maven.apache.org/guides/mini/guide-coping-with-sun-jars.html
[WARNING] The artifact javax.measure:jsr-275:jar:1.0.0 has been relocated to javax.measure:jsr-275:jar:0.9.1
[WARNING] The artifact org.apache.commons:commons-io:jar:1.3.2 has been relocated to commons-io:commons-io:jar:1.3.2: https://issues.sonatype.org/browse/MVNCENTRAL-244
[WARNING] The artifact junit:junit-dep:jar:4.11-beta-1 has been relocated to junit:junit:jar:4.11-beta-1: The artifact junit:junit does not contain Hamcrest anymore but declares a dependency to Hamcrest. Thus, junit:junit-dep has become obsolete.
^C[zgrinber@zgrinber poms_deps_with_no_ignore_long]$ date
**Sun Sep  3 14:50:15 IDT 2023**
```

![image](https://github.com/RHEcosystemAppEng/exhort-javascript-api/assets/75700623/3a55974e-49fb-4b81-ad28-c8a9eb7c7b70)

Reply:
@ruromero it appears that the infinite loop is not happening anymore when running the latest version of maven dependency plugin:
```shell
mvn -q org.apache.maven.plugins:maven-dependency-plugin:3.6.0:tree -Dverbose -DoutputFile=output.file
```